developing-without-a-build/article.md

```diff
@@ -0,0 +1,257 @@
+
+To handle these cases we can use `import.meta.url` to get information about the current module's location.
+
+`import.meta` is a special object which contains metadata about the currently executing module. `url` is the first property that's exposed here, and works a lot like `__dirname` in node js.
+
+`import.meta.url` points to the url the module was imported with:
+```js
+console.log(import.meta.url); // logs http://localhost:8080/path/to/my/file.js
+```
+
+We can use the `URL` API for easy URL building. For example to request a json file:
+```javascript
+const lang = 'en-US';
+
+// becomes http://localhost:8080/path/to/my/translations/en-US.json
+const translationsPath = new URL(`./translations/${lang}.json`, import.meta.url);
+
+const response = await fetch(translationsPath);
+```
+
+[Read more about import.meta at MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import.meta)
+
+### Loading other packages
+When building an application you will quickly run into having to include other packages from npm. This works just fine in the browser, for example let's install and use lodash:
+
+```bash
+npm i --save lodash-es
+```
+
+```javascript
+import kebabCase from '../node_modules/lodash-es/kebabCase.js';
+
+console.log(kebabCase('camelCase'));
+```
+
+Lodash is a very modular library, the `kebabCase` function depends on a lot of other modules. These dependencies are taken care of automatically, the browser resolves and imports them for you:
+
+![](./images/lodash-es.png)
+
+Writing explicit paths to your node modules folder is a bit unusual. While it works just fine, most people are used to writing what's called a bare import specifier:
+```javascript
+import { kebabCase } from 'lodash-es';
+import kebabCase from 'lodash-es/kebabCase.js';
+```
+
+This doesn't work in the browser out of the box, as it doesn't know what file to request when it sees this path. Luckily there is a [new browser API called import maps](https://github.com/WICG/import-maps) which lets you instruct the browser how to resolve these imports:
+```html
+<script type="importmap">
+  {
+    "imports": {
+      "lodash-es": "./node_modules/lodash-es/lodash.js",
+      "lodash-es/": "./node_modules/lodash-es/"
+    }
+  }
+</script>
+```
+
+It's currently [implemented in chrome behind a flag](https://developers.google.com/web/updates/2019/03/kv-storage#import_maps), and it's easy to shim on other browsers [with es-module-shims](https://www.npmjs.com/package/es-module-shims). Until we get broad browser support, that can be an interesting option during development.
+
+If you don't want to go quite as cutting edge but still want to use the bare imports a great option is to use a web server which rewrites the bare imports on the fly into explicit paths before serving them to the browser. There are some quite servers available that do this. Of course I recommend [es-dev-server](https://www.npmjs.com/package/es-dev-server) since I wrote it myself :)
```

Comment:
"Of course I recommend [es-dev-server](https://www.npmjs.com/package/es-dev-server) since I wrote it myself :)"
that is not a good reason for others use something :p
also it sounds like it's a one man show... lessens the value I my eyes 🙈

better be very explicit and explain why you use it or just mention that it's the one you are using in this blog post

Reply:
Good point... do you think we should not mention at all that we're promoting our own tool?

Reply:
say "Open Web Components recommends our `es-dev-server`. It's A, B, C, and D, which help you get things done" where A B C and D are advantages to our product